core/src/main/scala/org.broadinstitute.dsde.workbench.leonardo/models.scala

```diff
@@ -1,19 +1,146 @@
 package org.broadinstitute.dsde.workbench.leonardo
 
+import java.net.URL
 import java.time.Instant
 
+import cats.implicits._
 import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, parseGcsPath}
+import enumeratum.{Enum, EnumEntry}
+import ca.mrvisser.sealerate
 
-final case class MachineConfig(numberOfWorkers: Option[Int] = None,
-                               masterMachineType: Option[String] = None,
-                               masterDiskSize: Option[Int] = None, //min 10
-                               workerMachineType: Option[String] = None,
-                               workerDiskSize: Option[Int] = None, //min 10
-                               numberOfWorkerLocalSSDs: Option[Int] = None, //min 0 max 8
-                               numberOfPreemptibleWorkers: Option[Int] = None)
+import scala.collection.immutable
+import scala.util.matching.Regex
+
+
+final case class MachineType(value: String) extends AnyVal
```

Comment:
I think I added `MachineTypeName` in workbench-libs `google2` in https://github.com/broadinstitute/workbench-libs/pull/284… but I can consolidate as I integrate those changes.